datalake/src/main/java/com/sequenceiq/datalake/service/sdx/dr/SdxBackupRestoreService.java

```diff
@@ -695,4 +698,47 @@ private static boolean isVersionEqual(SdxCluster cluster, String baseVersion) {
         Comparator<Versioned> versionComparator = new VersionComparator();
         return versionComparator.compare(cluster::getRuntime, () -> baseVersion) == 0;
     }
+
+    public boolean isDatalakeInBackupOrRestore(String sdxClusterName, String actorCrn) {
+        if (sdxClusterName == null || actorCrn == null) {
+            LOGGER.error("Didn't find datalake name or actor CRN. Can not validate if the datalake is in backup/restore. Continue to next step.");
+            return false;
+        }
+        DatalakeBackupStatusResponse backupStatus;
+        DatalakeRestoreStatusResponse restoreStatus;
+        try {
+            backupStatus = datalakeDrClient.getBackupStatus(sdxClusterName, actorCrn);
+            restoreStatus = datalakeDrClient.getRestoreStatus(sdxClusterName, actorCrn);
```

Comment:
ok, backup is IN_PROGRESS, restore is not yet performed. Exception thrown, pattern matched (restore never happened), we report it is ok to go. But backup is ongoing. Is this right then?

Reply:
Thank you a lot @votez ! Updated those logics by separating backup and restore into different functions because otherwise it will have complexity error for checkStyle. Please let me know if it looks good!=)